docs/manual-EN/1.overview/3.design-and-architecture/2.storage-design.md

```diff
@@ -1 +1,136 @@
-<!-- TODO translate from CN>
\ No newline at end of file
+# Storage Design
+
+## Abstract
+
+This document is to give an introduction on the storage design of the graph database Nebula Graph.
+
+The Storage Service of nebula is composed of two parts, one is `Meta Service` that stores the meta data, the other is `Storage Service`that stores the data. The two services are in two independent processes. The data and deployment are separated as well. But their architectures are almost the same.
+
+## Architecture
+
+---
+![](https://oscimg.oschina.net/oscnet/4b44fadbb1ee0455df6ed635d99154d2333.jpg)
+**Picture One The Architecture of Storage Service**
+
+---
+
+As shown in Picture One, there are three layers in Storage Service. The bottom layer is the local storage engine, providing `get`, `put`, `scan` and `delete` operations on local data. The related interfaces are in `KVStore / KVEngine.h` and users can develop their own local store plugins based on their needs. Currently, nebula provides store engine based on RockDB.
+
+Above the local storage engine is the consensus layer that implements multi group raft. Each partition corresponding to a Raft group, is for the data sharding. Currently, nebula uses `hash` to shard data. When creating a space, users need to specify the partition number. Once set, partition number can not be changed. Generally, the partition number should meet the scale out needs in the future.
+
+Above the consensus layer is the storage interfaces that defines a set of APIs that are related with graph. These API requirements are translated into a set of kv operations targeting at the corresponding partition. It is this layer that makes the storage service a real graph storage, otherwise it's just a kv storage. Nebula doesn't separate kv as an independent service as graph query involves a lot of calculation that use schema, which is not existed in kv layer. Such architecture makes pushing down more easily.
+
+## Schema & Partition
+
+Nebula stores the vertices, edges and properties. Efficient property filtering is critical for a Graph Database.
+
+Nebula uses tags to indicate vertices types. One vertexID can relate with multiple tags, and each tag has its own properties. In kv storage, we use `vertexID + TagID` to represent key, and the related properties are placed behind value after encoding, the format is shown in Picture Two:
+
+---
+![](https://oscimg.oschina.net/oscnet/524d35b78a3fa8178536c9710f39631ad38.jpg)
+**Picture Two Vertex Key Format**
```

Comment:
The description of the picture should be put at the bottom of the picture and the words "Picture Two" and "Two Vertex Key Format" should be written in two styles, or they are mixed up.